pages/tutorials/graphql_sso.md.erb

```diff
@@ -0,0 +1,250 @@
+
+```graphql
+mutation EnableProvider {
+  ssoProviderEnable(
+    input: {
+      id: "<provider id>"
+    }
+  ) {
+    ssoProvider {
+      state
+      url
+    }
+  }
+}
+```
+
+You should now see that the provider’s state is enabled.
+
+See the `SSOProviderUpdatePayload` documentation for other properties that can be configured on your SSO provider, such as `sessionDurationInHours` and `note`.
+
+## Setting up SAML (Okta, OneLogin, ADFS and others)
+
+The first step in setting up a SAML-based provider is to use the `ssoProviderCreate` mutation to create a new provider in Buildkite and retrieve the details you’ll need for your SSO provider’s system.
+
+```graphql
+mutation CreateProvider {
+  ssoProviderCreate(input: {
+    organizationId: "<organization id>",
+    type: SAML
+  }) {
+    ssoProvider {
+      state
+      ... on SSOProviderSAML {
+        serviceProvider {
+          issuer
+          ssoURL
+          metadata {
+            url
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+The next step is to log into your SSO provider’s system and setup Buildkite using the `issuer`, `ssoURL` and `metadata` returned above. Once setup, your SSO provider should provide a metadata URL that you can give to Buildkite.
+
+Using the metadata URL returned from your SSO provider, call the `ssoProviderUpdate` mutation to have Buildkite automatically retrieve the details and set it up, ready for a test login. This will not yet affect any of your Buildkite users.
+
+```graphql
+mutation UpdateProviderMetaData {
+  ssoProviderUpdate(input: {
+    id: "<provider id>",
+    identityProvider: {
+      metadata: {
+        url: "<meta-data URL>"
+      }
+    }
+  }) {
+    ssoProvider {
+      state
+      url
+    }
+  }
+}
+```
+
+The final step before enabling the SSO provider in Buildkite, is to perform a test login.
+
+Perform a test login via your SSO provider, or using the `url` returned from the update mutation.
+
+Once a test login has been completed, you can do the final step: enabling the provider using the `ssoProviderEnable` mutation. Running this mutation will require all your users to login via your SAML provider before they can access your organization on Buildkite.
+
+```graphql
+mutation EnableProvider {
+  ssoProviderEnable(
+    input: {
+      id: "<provider id>"
+    }
+  ) {
+    ssoProvider {
+      state
+      url
+    }
+  }
+}
+```
+
+You should now see that the provider’s state is enabled.
+
+See the `SSOProviderUpdatePayload` documentation for other properties that can be configured on your SSO provider, such as `sessionDurationInHours` and `note`.
+
+## Finding an SSO provider’s details
+
+If you need to find the id of a particular SSO provider, you can query the `ssoProviders` field of your organization:
```

Comment:
What's the use case for this? I like the section, I just don't understand what people would be doing that makes them come to this page and click this heading.

Reply:
Ah. So mainly it's if you go off-piste, and you run another GraphQL query during the setup process and lose the response from a previous mutation. Because they're create/updates, you can't just re-run them… so you have to go and find the details (like the ID) again yourself mainly.

But instead of just the `id`, I just added a whole bunch of properties that are queried, for other situations as well, like seeing whether the provider is activated (which affects all your users), and other tasks like that.

If we had a proper SSO Settings page where you could manage it all yourself, we probably wouldn't need this section.